packages/components/Button/src/Button.tsx

```diff
@@ -70,17 +97,46 @@ export const Button = compose<ButtonType>({
         });
       }
       const label = accessibilityLabel ?? childText;
-
-      return (
-        <Slots.root {...mergedProps} accessibilityLabel={label}>
+      const buttonContent = (
+        <React.Fragment>
           {loading && <ActivityIndicator />}
           {shouldShowIcon && iconPosition === 'before' && <Slots.icon {...iconProps} />}
           {React.Children.map(children, (child) =>
             typeof child === 'string' ? <Slots.content key="content">{child}</Slots.content> : child,
           )}
           {shouldShowIcon && iconPosition === 'after' && <Slots.icon {...iconProps} />}
-        </Slots.root>
+        </React.Fragment>
       );
+
+      const hasRipple = Platform.OS === 'android';
+      if (hasRipple) {
+        const [extractedMargin, styleWithoutMargin] = extractMarginAndroid(mergedProps.style);
+        return (
+          <Slots.root style={extractedMargin}>
+            <Slots.ripple
+              accessibilityLabel={label}
+              {...mergedProps}
+              style={styleWithoutMargin}
+              onPressIn={() => {
```

Comment:
This should be a memoized function. Same with the function that's passed into onPressOut

However, don't use hooks, use memoize directly. We can't use hooks in the inner render as the inner render is conditionally called.

Reply:
I'm guessing this will cause our rerender tests to fail, so I would run our unit tests using yarn test